src/Lockedfield.php

```diff
@@ -68,6 +68,34 @@ public static function canCreate()
         return Session::haveRight(self::$rightname, UPDATE);
     }
 
+    public function canPurgeItem()
+    {
+        $item = new $this->fields['itemtype']();
+        if (
+            $item->getFromDB($this->fields['items_id']) //not a global lock
+            && $item->isEntityAssign()
+            && !Session::haveAccessToEntity($item->getEntityID(), $item->isRecursive()) // no access to main item entity
+        ) {
+            return false;
+        }
```

Comment:
If there is a lack of checks for the purge right, I guess there is the same lack of checks for the creation and  the update right. Could you fix that too ?